### PR TITLE
[stable/rocketchat] Add Affinities, podAntiAffinity and PodDisruptionBudget

### PR DESCRIPTION
**Pull request by stefanandres**
#### What this PR does / why we need it:

* RocketChat makes it possible to set a replica count > 1. When running with multiple replicas, it should be possible to:
  * Make sure they are running on different nodes (Affinities)
  * Prevent node evictions to trigger service outages (PodDisruptionBudget)

#### How to test

* Affinities
```yaml
# values.yaml:
podAntiAffinity: hard
```

```sh
kubectl scale deployments -l app.kubernetes.io/name=rocketchat --replicas=$HIGHER_THAN_NODE_COUNT
```
It should have multiple pods pending with and `kubectl describe pod` on the pending pods should tell something like `0/3 nodes are available: 3 node(s) didn't match pod affinity/anti-affinity, 3 node(s) didn't match pod anti-affinity rules.`

* PodDisruptionBudget
```yaml
replicaCount: 3
```

```sh
kubectl describe poddisruptionbudgets.policy

Status:
    Allowed disruptions:  2
    Current:              3
    Desired:              1
    Total:                3
```

#### Checklist
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/rocketchat/templates/poddisruptionbudget.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{{- if gt .Values.replicaCount 1.0 }}
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  labels:
+    app: {{ template "rocketchat.name" . }}
```
Comment (srueg) [inline stable/rocketchat/templates/poddisruptionbudget.yaml]: Please use the "new" style labels (i.e. `app.kubernetes.io/name`)

File: stable/rocketchat/values.yaml [hunk 2]
```diff
@@ -26,6 +27,32 @@ extraEnv:  # |
...
+## The value "" will disable pod anti-affinity so that no anti-affinity rules will be configured.
+##
+podAntiAffinity: ""
+
+## If anti-affinity is enabled sets the topologyKey to use for anti-affinity.
+## This can be changed to, for example, failure-domain.beta.kubernetes.io/zone
+##
+podAntiAffinityTopologyKey: kubernetes.io/hostname
+
+## Assign custom affinity rules to the alertmanager instance
```
Comment (srueg) [inline stable/rocketchat/values.yaml]: Should probably say `RocketChat` instead of `alertmanager`

**Review by srueg — CHANGES_REQUESTED** [review]

File: stable/rocketchat/templates/poddisruptionbudget.yaml (hunk 1 above)
Comment (stefanandres) [reply stable/rocketchat/templates/poddisruptionbudget.yaml]: Ah, thanks - fixed

File: stable/rocketchat/values.yaml (hunk 2 above)
Comment (stefanandres) [reply stable/rocketchat/values.yaml]: Yep, fixed that too now

File: stable/rocketchat/README.md [hunk 3]
```diff
@@ -53,6 +53,10 @@ Parameter | Description | Default
 `smtp.host` | Hostname of the SMTP server | `""`
 `smtp.port` | Port of the SMTP server | `587`
 `extraEnv` | Extra environment variables for Rocket.Chat. Used with `tpl` function, so this needs to be a string | `""`
+`podAntiAffinity` | Pod anti-affinity can prevent the scheduler from placing RocketChat replicas on the same node. The default value "soft" means that the scheduler should *prefer* to not schedule two replica pods onto the same node but no guarantee is provided. The value "hard" means that the scheduler is *required* to not schedule two replica pods onto the same node. The value "" will disable pod anti-affinity so that no anti-affinity rules will be configured. | `""` |
+`podAntiAffinityTopologyKey` | If anti-affinity is enabled sets the topologyKey to use for anti-affinity. This can be changed to, for example `failure-domain.beta.kubernetes.io/zone`| `kubernetes.io/hostname` |
+| `affinity` | Assign custom affinity rules to the prometheus instance https://kubernetes.io/docs/concepts/configuration/assign-pod-node/ | `{}` |
```
Comment (srueg) [inline stable/rocketchat/README.md]: Another typo `prometheus`

**Review by srueg — CHANGES_REQUESTED** [review]

**Comment by srueg** [comment]
How do you guys feel about setting a default for `podAntiAffinity`?
I'd propose to set the default to `hard`. This might hinder deployments on small clusters, but since the default `replicaCount` is `1` this shouldn't be a problem.

File: stable/rocketchat/README.md (hunk 3 above)
Comment (stefanandres) [reply stable/rocketchat/README.md]: Thanks, fixed

**Comment by stefanandres** [comment]
> How do you guys feel about setting a default for `podAntiAffinity`?
> I'd propose to set the default to `hard`. This might hinder deployments on small clusters, but since the default `replicaCount` is `1` this shouldn't be a problem.

Personally I'd always configure hard. I've used the same defaults as the prometheus-operator helm chart (which is "").

The majority of use cases should be to scale along side the number of nodes in a cluster (hence using hard would be the best option).

**Comment by stefanandres** [comment]
@srueg What can I do to get this merged?

**Comment by srueg** [comment]
Sorry, totally forgot about it. Thanks for your PR :)

/approve

**Comment by stefanandres** [comment]
To my understanding, someone also has to /lgtm to get this merged, correct? Can you do that?  :)

**Pull request merged**